### PR TITLE
Fixed favorites and recent nodes selecting wrong node

### LINKED ISSUE #6535: Problem with new favorites panel in add node dialog
**Operating system or device - Godot version:**
Windows 10, Godot https://github.com/godotengine/godot/commit/0e962311be7fa5565a456ecd328202c2d6f6ca13

**Issue description** (what happened, and what was expected):
When add "Node" to favorites in add node dialog, after selected it, on the list is selected wrong node, "GraphNode"  instead of "Node":

### PR DESCRIPTION
**Pull request by Piet-G**
Fixed favorites and recent nodes selecting wrong node.

Fixes #6535

### CODE REVIEW

File: editor/create_dialog.cpp [hunk 1]
```diff
@@ -246,9 +246,9 @@ void CreateDialog::add_type(const String &p_type, HashMap<String, TreeItem *> &p
 		to_select_type = to_select_type.split(" ")[0];
 		bool current_item_is_preffered;
 		if (cpp_type) {
-			current_item_is_preffered = ClassDB::is_parent_class(p_type, preferred_search_result_type) && !ClassDB::is_parent_class(to_select_type, preferred_search_result_type);
+			current_item_is_preffered = ClassDB::is_parent_class(p_type, preferred_search_result_type) && !ClassDB::is_parent_class(to_select_type, preferred_search_result_type) && search_box->get_text() != to_select_type;
```
Comment (akien-mga) [inline editor/create_dialog.cpp]: Could you also fix the typo with `current_item_is_preffered`? (should be `preferred`)

Comment (DualMatrix) [reply editor/create_dialog.cpp]: Fixed

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Now that Godot 3.1 has been released, we only plan to cherry-pick critical bug fixes to the 3.0 branch, unless there is very strong support in favor of a specific non-critical issue. Removing cherrypick label for 3.0.

**Pull request merged**